### PR TITLE
Refactor Document class to reduce calls to marklogic for the metadata fields

### PR DESCRIPTION
**Pull request by anthonyhashemi**
## Changes in this PR:
- Refactor Document class to reduce calls to marklogic for the metadata
-  In particular updated `name`, `court`, `document_date_as_string` and `document_date_as_date` (previously `judgment_date_...`) on Document class and `neutral_citation` on Judgment class.
- Added `xml_etree` cached property to Document class and adjusted content_as_html to be a cached_property also.
- Removed `get_judgment_name, get_judgment_citation, get_judgment_court, get_judgment_work_date` from the Client class and associated `.xqy` files.

## Trello card / Rollbar error (etc)
https://trello.com/c/gEGuXxmq/1156-api-client-refactor-the-apiclient-document-class-to-store-xml-rather-than-querying-the-db-for-each-metadata

### CODE REVIEW

File: src/caselawclient/models/documents.py [hunk 1]
```diff
@@ -78,20 +80,35 @@ def public_uri(self) -> str:
...
+                "akn": "http://docs.oasis-open.org/legaldocml/ns/akn/3.0",
+            },
+        )
 
     @cached_property
-    def judgment_date_as_string(self) -> str:
-        return self.api_client.get_judgment_work_date(self.uri)
+    def document_date_as_string(self) -> str:
+        return self._get_xpath_match_string(
+            "/root/akn:akomaNtoso/akn:judgment/akn:meta/akn:identification/akn:FRBRWork/akn:FRBRdate/@date",
```
Comment (dragon-dxw) [inline src/caselawclient/models/documents.py:101]: `akn:*` not `akn:judgment` ?

File: tests/models/test_documents.py [hunk 2]
```diff
@@ -38,31 +38,67 @@ def test_document_exists_check(self, mock_api_client):
...
+            </root>
+        """
 
         document = Document("test/1234", mock_api_client)
 
-        assert document.name == "Test Judgment v Test Judgement"
-        mock_api_client.get_judgment_name.assert_called_once_with("test/1234")
+        assert document.name == "Test Claimant v Test Defendant"
+        mock_api_client.get_judgment_xml.assert_called_once_with(
+            "test/1234", show_unpublished=True
```
Comment (dragon-dxw) [inline tests/models/test_documents.py:51]: wary that getting the judgment name might automatically show unpublished judgment's names -- I assume this is caught elsewhere?

**Review by dragon-dxw — APPROVED** [review]
Comments on the xquery not working for all document types might be best left for another pass.

File: src/caselawclient/models/documents.py [hunk 3]
```diff
@@ -78,20 +80,35 @@ def public_uri(self) -> str:
...
+        return self._get_xpath_match_string(
+            "//akn:FRBRname/@value",
+            {"akn": "http://docs.oasis-open.org/legaldocml/ns/akn/3.0"},
+        )
 
     @cached_property
     def court(self) -> str:
-        return self.api_client.get_judgment_court(self.uri)
+        return self._get_xpath_match_string(
+            "/root/akn:akomaNtoso/akn:judgment/akn:meta/akn:proprietary/uk:court/text()",
```
[the remark this answers is not in the document]
Comment (anthonyhashemi) [reply src/caselawclient/models/documents.py:91]: will be done in separate PR for fixing for press summaries

File: src/caselawclient/models/documents.py (hunk 1 above)
Comment (anthonyhashemi) [reply src/caselawclient/models/documents.py:101]: likewise

File: tests/models/test_documents.py (hunk 2 above)
Comment (anthonyhashemi) [reply tests/models/test_documents.py:51]: im assuming we just trust that once we have got the document we are happy that we can use it and all of its data.

**Pull request merged**